### ISSUE TITLE
ClearML logger (ex Allegro Trains AI)

### ISSUE THREAD

**Issue by TezRomacH** [PyTorchLightning/pytorch-lightning#5921]
### Motivation

ClearML is opensourced self-hosted service for MLOps, logging and experiment management.
https://github.com/allegroai/clearml/ 

Previously known as `Allegro Trains AI`.

Their repository has more than 2k stars on GitHub and very active community.

### Pitch

I would like to have an implementation for `pytorch_lightning.loggers.ClearMLLogger`.

```python
from pytorch_lightning import Trainer
from pytorch_lightning.loggers import ClearMLLogger
clearml_logger = ClearMLLogger(
    project_name="default",
    task_name="train"
)
trainer = Trainer(logger=clearml_logger)
```

**Comment by jzazo** [comment]
Why did you close? Is there a logger already available? (apparently an old one existed, but cannot find it in newer PL versions).
Also, it is not clear from ClearML documentation how to log tables/metrics/figures/etc using PL.

**Comment by TezRomacH** [comment]
I closed id, cuz i have found that ClearML is already compatible with PL 

https://github.com/allegroai/clearml/blob/master/examples/frameworks/pytorch-lightning/pytorch_lightning_example.py

**Comment by jzazo** [comment]
If you do `self.log("var", var)` inside a PL model, will it work? I imagined that if would record gradients and other simulation associated metrics. Can you confirm if you have tried? Thanks.

**Comment by TezRomacH** [comment]
it wont :( 

But, i suppose you might try to get logger
```
task = Task.init(project_name="examples", task_name="hydra configuration")
logger = task.get_logger()
```

and use it as `logger.report_histogram(...)` inside lightning model
https://github.com/allegroai/clearml/blob/912f6f5ba2328b26de042de03f02de5802df360f/clearml/logger.py

**Comment by jzazo** [comment]
Thanks. The ClearML logger is a nice alternative, but a Lightning logger is still necessary to interface with the PL logging tools. Maybe consider reopening your issue? I think it would be worth having such a logger.

**Comment by dmarx** [comment]
Looks like ClearML integrates with tensorboard. Have you tried using a `TensorboardLogger`?

**Comment by melihozaydin** [comment]
Any updates on this?
